torch/_inductor/fx_passes/reinplace.py

```diff
@@ -176,7 +176,11 @@ def _decompose_scatter_mutating(
 
 def scatter_always_uses_mutation(node: torch.fx.Node) -> bool:
     _, _, view_ops = node.args
-    return any(view.target in _ALWAYS_MUTATING_SCATTER_OPS for view in view_ops)  # type: ignore[union-attr]
+    return any(
+        t in _ALWAYS_MUTATING_SCATTER_OPS
+        for view in view_ops  # type: ignore[union-attr]
+        if isinstance(t := view.target, torch._ops.OpOverload)  # type: ignore[union-attr]
```

Comment:
Quibble: adding the intermediate `t` variable doesn't make the code any clearer.

----

FYI: I don't think this works well here but I've been using `TypeGuard` for similar errors in loops, [like this](https://github.com/pytorch/pytorch/blob/e59188cd5c65b35a6437c90fa062aeb25278d268/torch/_inductor/ir.py#L4922-L4925)